flexmeasures/data/models/assets.py

```diff
@@ -39,10 +39,14 @@ class AssetType(db.Model):
     yearly_seasonality = db.Column(db.Boolean(), nullable=False, default=False)
 
     def __init__(self, **kwargs):
-        generic_asset_type = GenericAssetType(
-            name=kwargs["name"], description=kwargs.get("hover_label", None)
-        )
-        db.session.add(generic_asset_type)
+        generic_asset_type = GenericAssetType.query.filter_by(
+            name=kwargs["name"]
+        ).one_or_none()
+        if not generic_asset_type:
+            generic_asset_type = GenericAssetType(
+                name=kwargs["name"], description=kwargs.get("hover_label", None)
+            )
+            db.session.add(generic_asset_type)
```

Comment:
I only needed to do this because tests were failing (when a GenericAssetType with the same name already existed). It is useful for something now, but I don't see it as an essential part of application code.

- I would only add this in MarketType or WeatherSensorType if we need it, in the last time we still have those.
- I haven't looked at project 9 for this.